src/queries.mjs

```diff
@@ -0,0 +1,150 @@
+      (_, __, inner) => inner
+    );
+
+    const metadata = parserUtils.parseYAMLIntoMetadata(sanitizedString);
+
+    apiEntryMetadata.updateProperties(metadata);
+  };
+
+  /**
+   * Transforms plain reference to Web/JavaScript/Node.js types
+   * into Markdown links containing the proper reference to said types
+   *
+   * @param {import('unist').Node} node The Type Node
+   */
+  const updateTypeToReferenceLink = node => {
+    const parsedReference = node.value.replace(
+      createQueries.QUERIES.normalizeTypes,
+      parserUtils.transformTypeToReferenceLink
+    );
+
+    node.type = 'html';
+    node.value = parsedReference;
+  };
+
+  /**
+   * Parse a Heading Node into Metadata and updates the current Metadata
+   *
+   * @param {import('unist').Node} node The Heading Node
+   * @param {ReturnType<import('./metadata.mjs').default>} apiEntryMetadata The API entry Metadata
+   */
+  const addHeadingMetadata = (node, apiEntryMetadata) => {
+    const heading = transformNodesToString(node.children);
+
+    const parsedHeading = parserUtils.parseHeadingIntoMetadata(
+      heading,
+      node.depth
+    );
+
+    apiEntryMetadata.setHeading(parsedHeading);
+  };
+
+  /**
+   * Updates a Markdown Link into a HTML Link for API Docs
+   *
+   * @param {import('unist').Node} node Thead Link Node
+   */
+  const updateMarkdownLink = node => {
+    node.url = node.url.replace(
+      createQueries.QUERIES.markdownUrl,
+      (_, filename, hash = '') => `${filename}.html${hash}`
+    );
+  };
+
+  /**
+   * Updates a Markdown Link Reference into an actual Link to the Definition
+   *
+   * @param {import('unist').Node} node Thead Link Reference Node
+   * @param {Array<import('unist').Node>} definitions The Definitions of the API Doc
+   */
+  const updateLinkReference = (node, definitions) => {
+    const definition = definitions.find(
+      ({ identifier }) => identifier === node.identifier
+    );
+
+    node.type = 'link';
+    node.url = definition.url;
+  };
+
+  /**
+   * Parses a Stability Index Entry and updates the current Metadata
+   *
+   * @param {import('unist').Node} node Thead Link Reference Node
+   * @param {ReturnType<import('./metadata.mjs').default>} apiEntryMetadata The API entry Metadata
+   */
+  const addStabilityIndexMetadata = (node, apiEntryMetadata) => {
+    const stabilityIndexString = transformNodesToString(
+      node.children[0].children
+    );
+
+    const stabilityIndex =
+      createQueries.QUERIES.stabilityIndex.exec(stabilityIndexString);
+
+    apiEntryMetadata.updateProperties({
+      stability_index: {
+        index: Number(stabilityIndex[1]),
+        description: stabilityIndex[2].replaceAll('\n', ' ').trim(),
+      },
+    });
+  };
+
+  return {
+    addYAMLMetadata,
+    updateTypeToReferenceLink,
+    addHeadingMetadata,
+    updateMarkdownLink,
+    updateLinkReference,
+    addStabilityIndexMetadata,
+  };
+};
+
+// This defines the actual REGEX Queries
+createQueries.QUERIES = {
```

Comment:
I think we can assign it as a constant under `createQueries` and return it at the bottom. Using `QUERIES.something` instead of `createQueries.QUERIES.something` seems more readable

For example;
```mjs
const createQueries = () => {
  const QUERIES = {
    markdownUrl: /^(?![+a-z]+:)([^#?]+)\.md(#.+)?$/i,
    // ... other queries
  };

  const addYAMLMetadata = (node, apiEntryMetadata) => {
    const sanitizedString = node.value.replace(
      QUERIES.yamlInnerContent,
      (_, __, inner) => inner
    );

    const metadata = parserUtils.parseYAMLIntoMetadata(sanitizedString);

    apiEntryMetadata.updateProperties(metadata);
  };

  // ... other methods

  const UNIST = {
    isStabilityIndex: ({ type, children }) =>
      type === 'blockquote' &&
      QUERIES.stabilityIndex.test(transformNodesToString(children)),
      // ... other unists
  };
  
  return {
    addYAMLMetadata,
    // ... other methods
    UNIST,
    QUERIES,
  };
};
```

It will also be necessary to update the usage in the `parser.mjs` file 👀

Reply:
I don't think we need this, the current solution is fine and it makes it obvious that these are directly related to the usage of createQueries